### PR TITLE
DSP/HLE: Implement mixer processing

### PR DESCRIPTION
**Pull request by MerryMage**
Includes audio output.

Only implemented well-understood portions of the firmware.

### CODE REVIEW

File: src/audio_core/hle/mixers.cpp [hunk 1]
```diff
@@ -0,0 +1,203 @@
+// Copyright 2016 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
```
Comment (lioncash) [inline src/audio_core/hle/mixers.cpp]: This isn't needed in the cpp file

File: src/audio_core/hle/mixers.cpp [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    if (state.mixer2_enabled) {
+        for (size_t sample = 0; sample < samples_per_frame; sample++) {
+            for (size_t channel = 0; channel < 4; channel++) {
+                state.intermediate_mix_buffer[2][sample][channel] = read_samples.mix2.pcm32[channel][sample];
+            }
+        }
+    }
+}
+
+void Mixers::AuxSend(IntermediateMixSamples& write_samples, const std::array<QuadFrame32, 3> input) {
```
Comment (lioncash) [inline src/audio_core/hle/mixers.cpp]: was `input` supposed to be a const reference?

Comment (MerryMage) [inline src/audio_core/hle/mixers.cpp]: Yes

**Comment by MerryMage** [comment]
Thanks @lioncash.

File: src/audio_core/hle/dsp.cpp [hunk 3]
```diff
@@ -75,17 +121,12 @@ void Shutdown() {
 }
 
 bool Tick() {
-    SharedMemory& read = ReadRegion();
-    SharedMemory& write = WriteRegion();
+    StereoFrame16 current_frame = {};
 
-    std::array<QuadFrame32, 3> intermediate_mixes = {};
+    // TODO(merry): Only perform audio configuration when emulated application signals us to do so.
```
Comment (Subv) [inline src/audio_core/hle/dsp.cpp]: What does this comment mean?

Comment (MerryMage) [inline src/audio_core/hle/dsp.cpp]: Would something like "TODO: Check dsp::DSP semaphore" be clearer?

Comment (Subv) [inline src/audio_core/hle/dsp.cpp]: Not really, that still doesn't explain why you have to check it

Comment (MerryMage) [inline src/audio_core/hle/dsp.cpp]: Ah. "TODO: Check dsp::DSP semaphore (which indicates emulated application has finished writing to shared memory region)"

Comment (Subv) [inline src/audio_core/hle/dsp.cpp]: That's better

**Comment by MerryMage** [comment]
Addressed @Subv's comment above.

**Comment by MerryMage** [comment]
Squash

**Comment by bunnei** [comment]
LGTM

**Pull request merged**